Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -672,7 +672,12 @@ namespace AZ
         // Execute user.cfg after modules have been loaded but before processing any command-line overrides
         AZ::IO::FixedMaxPath platformCachePath;
         m_settingsRegistry->Get(platformCachePath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_CacheRootFolder);
-        m_console->ExecuteConfigFile((platformCachePath / "user.cfg").Native());
+
+        AZ::IO::FixedMaxPath userConfigFilePath = platformCachePath / "user.cfg";
+        if (AZ::IO::SystemFile::Exists(userConfigFilePath.Native().c_str()))
+        {
+            m_console->ExecuteConfigFile(userConfigFilePath.Native());
+        }
```

Comment:
There is still a gap here that it doesn't use the FileIOBase to check for file existence, so if the user.cfg file is in the Engine.Pak in the GameLauncher it will not be found.

Reply:
We can add a existence check using both FileIOBase if it exists or SystemFile.
```suggestion
        if (auto fileIoBase = AZ::FileIOBase::GetInstance(); AZ::IO::SystemFile::Exists(userConfigFilePath.c_str())
            || (fileIoBase != nullptr && fileIoBase->Exists(userConfigFilePath.c_str())))
        {
            m_console->ExecuteConfigFile(userConfigFilePath.Native());
        }
```

Reply:
So use FileIO::Instance() singleton?

Reply:
Looks like the better solution is to handle this in 'ExecuteConfigFile' instead

Reply:
I don't think `ExecuteConfigFile` should handle it, because if it does it should log that it couldn't find the file to load, at which point the same issue has just been pushed up from the `SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile` to `Console::ExecuteConfigFile`